addons/website_sale/static/src/scss/website_sale.scss

```diff
@@ -278,6 +278,13 @@ $o-wsale-products-layout-grid-gutter-width: min($grid-gutter-width / 2, $o-wsale
                         background-color: $white !important;
                     }
                 }
+                .o_ribbon_right {
+                    @include o-position-absolute($top: 0, $left: 0);
+                    transform: translateX(calc(0.708 * (100% - #{2 * $ribbon-padding}) - 100%)) rotate(-45deg) translateX($ribbon-padding);
+                }
+                .o_tag_right {
+                    @include o-position-absolute($top: 0, $left: 0);
+                }
```

Comment:
If we do that, then at least try to share the code and modify the option to not say "Position" but "Position (grid view)" or a tooltip "Only on grid view". Otherwise we'll have a bug report for "it is not on the right". However, I see in grid view, the image size is fixed, can't we put those on the right of the image?

Reply:
Actually, there is a possibility to move the span with the ribbon in the card-body with the image in the product template but Im not sure if we can do that in V14.0.

Reply:
Yes, we can't move the ribbon in stable. But I was not speaking about moving it, just said the image has a fixed sized, known in scss. So you should be able to position the ribbon on its right with its current position in the DOM I think?

Reply:
![FireShot Capture 651 - Capture d’écran de 2021-02-22 14-40-29 png (3840×1080) - ](https://user-images.githubusercontent.com/52911687/108824220-c3db5000-75c1-11eb-94aa-601e482767e2.jpg)

As you can see here it's impossible because the ribbon is a rectangle cropped by its parent element (and therefore the only solution is to move it so that the parent element will be the card-body containing the image)

Reply:
Ok, then let's go for css sharing + warning on the option

Reply:
@bvr-odoo Still have this in my pipeline, any update?

Reply:
@qsm-odoo not yet but I could work on it tomorow if needed